python/sqlcommenter-python/google/cloud/sqlcommenter/sqlalchemy/executor.py

```diff
@@ -54,7 +55,10 @@ def before_cursor_execute(conn, cursor, sql, parameters, context, executemany):
         # folks using it in a web framework such as flask will
         # use it in unison with flask but initialize the parts disjointly,
         # unlike Django which uses ORMs directly as part of the framework.
-        data.update(get_flask_info())
+        info = get_flask_info()
+        if not info:
+            info = get_fastapi_info()
+        data.update(info)
```

Comment:
I am a little concerned on the scalability of this solution. If we add more frameworks, there will be more if and elseif statements added. Is it possible to think of an approach to use a switch-case or map based solution something like:

```
switch (get_framework()) {
  case "FastAPI":
     return get_fast_api_info()
  case "Flask":
    return get_flask_info()
  default:
   return None
}
```

Reply:
The only way we can do something like that is adding which framework are we using in `BeforeExecuteFactory`, as it should just work without configuration I don't know if that will be the better choice

Reply:
Thanks, refactoring into separate method looks better